dvc/repo/metrics/show.py

```diff
@@ -234,14 +235,20 @@ def _read_metrics(self, metrics, branch):
                 branch=branch,
             )
         else:
-            with self.tree.open(out.path) as fd:
-                metric = _read_metric(
-                    fd,
-                    typ=typ,
-                    xpath=xpath,
-                    rel_path=out.rel_path,
-                    branch=branch,
-                )
+            try:
+                with self.tree.open(out.path) as fd:
+                    metric = _read_metric(
+                        fd,
+                        typ=typ,
+                        xpath=xpath,
+                        rel_path=out.rel_path,
+                        branch=branch,
+                    )
+            except IOError as e:
+                if e.errno == errno.ENOENT:
+                    metric = "COULD NOT READ"
```

Comment:
This doesn't match ENOENT error, which means that there is no such file and your "COULD NOT READ" might be interpreted as formatting issue. Formatting problems are captured in the `_read_metric` itself.

Also, take a look at `_read_metrics_filesystem`, which simply returns None if metric doesn't exist. It is crutial that they both behave the same. Should we ignore it here as well?

Reply:
I think that returning None is ok, but I think that it would be good to somehow pass information that file could not be opened. None as value can be interpreted in many ways. Though, in such case we should probably inform user upon `_read_metrics_filesystem`